### ISSUE TITLE
violation of SOLID principle. Can we do it in more effective way?

### ISSUE THREAD

**Issue by sheremeta2018** [Andreeey/gotest#2]
From my perspective some Queue  will more affective here 
https://github.com/Andreeey/gotest/blob/4ef5d6c0fc3b31162b5680e8ee2c2322a6918cf8/service/parser/parser/parser.go#L65

**Comment by Andreeey** [comment]
Yes I must agree this breaks single responsibility principle. But in a lack of time it was made this way. Should be moved to some layer that is responsible for data transfer to micro-service or some queue indeed

**Comment by sheremeta2018** [comment]
Could you please improve this part with goroutines?

**Comment by sheremeta2018** [comment]
Don't worry. Just propose your approach to do it on the interview.

**Comment by Andreeey** [comment]
Ok. But anyway, refactored parser to use goroutines.
